### PR TITLE
Make Class:MaterialProcessing

### PR DESCRIPTION
**Pull request by anastasiyaprymolenna**
This PR contains changes to nmdc.yaml and the dependent data examples

MaterialProcessing class has been created.
BiosampleProcessing class has been deleted

### CODE REVIEW

**Comment by mslarae13** [comment]
https://github.com/microbiomedata/nmdc-schema/issues/1513
https://github.com/microbiomedata/nmdc-schema/issues/1514
https://github.com/microbiomedata/nmdc-schema/issues/1518

**Comment by mslarae13** [comment]
https://github.com/microbiomedata/nmdc-schema/issues/1519

**Comment by brynnz22** [comment]
This shouldn't require a migration since the name of the parent class (which is not used in Mongo as a collection) is changing. I'm going to remove the lable, but please correct me if I'm wrong!

 I'm not sure if the new type code needs to be allowed like what @aclum did here: https://github.com/microbiomedata/berkeley-schema-fy24/pull/61#event-11383230487.

**Comment by mslarae13** [comment]
@brynnz22 I think we still need to migrate the data. I believe some information is captured on the removed BiosampleProcessing class?

File: src/schema/nmdc.yaml [hunk 1]
```diff
@@ -1524,24 +1517,27 @@ classes:
...
     slots:
       - has_input
+    notes: 
+      - This class is a replacement for BiosampleProcessing.
     slot_usage:
       id:
         required: true
         structured_pattern:
-          syntax: "{id_nmdc_prefix}:bsmprc-{id_shoulder}-{id_blade}{id_version}{id_locus}"
+          syntax: "{id_nmdc_prefix}:matprc-{id_shoulder}-{id_blade}{id_version}{id_locus}"
```
Comment (aclum) [inline src/schema/nmdc.yaml:1533]: Do we want to make this class abstract? Looks similar to DataGeneration where we don't intend to use the parent class.

**Comment by aclum** [comment]
I think this is okay, at least on the sequencing side b/c we don't use BiosampleProcessing directly anywhere in mongo. The collections are for the subclasses, LibraryPreparation and Pooling.

**Pull request merged**